lib/edifunct/tokenizer.rb

```diff
@@ -27,6 +27,9 @@ def as_segments(message_as_string)
 
         Segment.new(tag: segment_tag, raw_segment: raw_segment, data_elements: data_elements)
       end
+        .reject do |segment|
```

Comment:
`UNA` shouldn't really be considered a segment at all, so the rejection/ignore step should happen earlier IMO. See this alternative PR here: https://github.com/orhantoy/edifunct/pull/6